stargz/stargz.go

```diff
@@ -677,6 +680,14 @@ func (w *Writer) AppendTar(r io.Reader) error {
 			// duplicated entries in the resulting layer.
 			continue
 		}
+
+		var xattrs map[string][]byte
+		if h.Xattrs != nil {
+			xattrs = make(map[string][]byte)
+			for k, v := range h.Xattrs {
+				xattrs[k] = []byte(v)
```

Comment:
What's your intention with this []byte(...) here? If you want to make a defensive copy, this won't do it. Or was this because it used to be string?

Reply:
`h.Xattrs` is from archive/tar.Header and it is defined as `map[string]string`

Reply:
Ah, sorry.